address/args.go

```diff
@@ -6,10 +6,12 @@ import (
 
 type addressOptions struct {
 	algorithm *xc.SignatureType
+	contract  *xc.ContractAddress
 }
 
 type AddressOptions interface {
 	GetAlgorithmType() (xc.SignatureType, bool)
+	GetContract() (xc.ContractAddress, bool)
```

Comment:
I think we should remove this, as it's not reasonable for Treasury product to generate addresses with a token address argument.  It should be refactored so that `TxBuilder` will format token addresses as needed.  The public key should always be passed as it's a signing address (public key is always known).